overlord/state/state.go

```diff
@@ -482,6 +491,58 @@ func (s *State) GetMaybeTimings(timings interface{}) error {
 	return nil
 }
 
+// AddTaskStatusChangedObserver adds a callback function that will be invoked
+// whenever tasks change status.
+// NOTE: Callbacks registered this way may be invoked in the context
+// of the taskrunner, so the callbacks should be as simple as possible, and return
+// as quickly as possible, and should avoid the use of i/o code or blocking, as this
+// will stop the entire task system.
+func (s *State) AddTaskStatusChangedObserver(f func(t *Task, old, new Status)) int {
```

Comment:
for clarity it would be good to name the result (id int)

Reply:
not sure the name of these methods still need "Observer" in them now, same for the other ones

Reply:
Done!